app/workers/migrations.php

```diff
@@ -0,0 +1,315 @@
+        $generatedSecret = bin2hex(\random_bytes(128));
+
+        $key = new Document([
+            '$id' => ID::unique(),
+            '$permissions' => [
+                Permission::read(Role::any()),
+                Permission::update(Role::any()),
+                Permission::delete(Role::any()),
+            ],
+            'projectInternalId' => $project->getInternalId(),
+            'projectId' => $project->getId(),
+            'name' => 'Transfer API Key',
+            'scopes' => [
+                'users.read',
+                'users.write',
+                'teams.read',
+                'teams.write',
+                'databases.read',
+                'databases.write',
+                'collections.read',
+                'collections.write',
+                'documents.read',
+                'documents.write',
+                'buckets.read',
+                'buckets.write',
+                'files.read',
+                'files.write',
+                'functions.read',
+                'functions.write',
+            ],
+            'expire' => null,
+            'sdks' => [],
+            'accessedAt' => null,
+            'secret' => $generatedSecret,
+        ]);
+
+        $consoleDB->createDocument('keys', $key);
+        $consoleDB->deleteCachedDocument('projects', $project->getId());
+
+        return $key;
+    }
+
+    /**
+     * Process Migration
+     *
+     * @return void
+     */
+    protected function processMigration(): void
+    {
+        /**
+         * @var Document $migrationDocument
+         * @var Transfer $transfer
+         */
+        $migrationDocument = null;
+        $transfer = null;
+        $projectDocument = $this->getConsoleDB()->getDocument('projects', $this->args['project']['$id']);
+        $tempAPIKey = $this->generateAPIKey($projectDocument);
+
+        try {
+            $migrationDocument = $this->dbForProject->getDocument('migrations', $this->args['migration']['$id']);
+            $migrationDocument->setAttribute('stage', 'processing');
+            $migrationDocument->setAttribute('status', 'processing');
+            $this->updateMigrationDocument($migrationDocument, $projectDocument);
+
+            $source = $this->processSource($migrationDocument->getAttribute('source'), $migrationDocument->getAttribute('credentials'));
+
+            $source->report();
+
+            $destination = new DestinationsAppwrite(
+                $projectDocument->getId(),
+                'http://appwrite/v1',
+                $tempAPIKey['secret'],
+            );
+
+            $transfer = new Transfer(
+                $source,
+                $destination
+            );
+
+            $migrationDocument->setAttribute('stage', 'source-check');
+            $this->updateMigrationDocument($migrationDocument, $projectDocument);
+
+            $migrationDocument->setAttribute('stage', 'destination-check');
+            $this->updateMigrationDocument($migrationDocument, $projectDocument);
```

Comment:
you're updating the status here to `destination-check` but I dont see any code that does it 🤔